app/services/legacy/dataset_import_service.rb

```diff
@@ -36,47 +37,76 @@ def update_or_create_dataset
     dataset.location1 = build_location
     dataset.licence = build_licence
     dataset.licence_other = build_licence_other
-    old_theme = legacy_dataset["theme-primary"]
-    secondary_theme = legacy_dataset["theme-secondary"]
-    dataset.theme_id = themes_cache.fetch(old_theme, nil)
-    dataset.secondary_theme_id = themes_cache.fetch(secondary_theme, nil)
+    dataset.theme_id = build_theme_id
+    dataset.secondary_theme_id = build_secondary_theme_id
     dataset.status = "published"
     dataset.save!(validate: false)
   end
 
-  def create_resources(dataset)
-    create_datafiles(dataset)
-    create_documents(dataset)
+  def update_or_create_dataset
+    if dataset.persisted?
+      dataset.update_columns(dataset_attributes)
```

Comment:
... ah ... I think this is what is speeding up the process? It's making a direct call to the db